src/plugins/expression-builder/eb-node.component.ts

```diff
@@ -1,10 +1,53 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, EventEmitter, Output, OnInit, ElementRef, OnDestroy } from '@angular/core';
 import { Node } from './model/node';
+import { EbNodeService } from './eb-node.service';
 
 @Component({
 	selector: 'q-grid-eb-node',
 	templateUrl: './eb-node.component.html'
 })
-export class EbNodeComponent {
-	@Input() public model: Node;
+export class EbNodeComponent implements OnInit, OnDestroy {
+	@Input() model: Node;
+	@Input() parent: EbNodeComponent;
+
+	self = this;
+	element: HTMLElement;
+	children = new Array<EbNodeComponent>();
```

Comment:
consider to move initialization to constructor and simplify it: `this.children = [];`

Reply:
what is point to make like this? if you dont like it i can do `children: Array<EBNodeComponent> = []` :)